xbmc/interfaces/json-rpc/PVROperations.cpp

```diff
@@ -337,6 +337,114 @@ JSONRPC_STATUS CPVROperations::GetTimerDetails(const std::string &method, ITrans
   return OK;
 }
 
+CFileItemPtr CPVROperations::GetBroadcastFromBroadcastid(unsigned int broadcastid)
+{
+  CFileItemPtr broadcast;
+  EpgSearchFilter filter;
+  filter.Reset();
+  filter.m_iUniqueBroadcastId = broadcastid;
+
+  CFileItemList broadcasts;
+  int resultSize = g_EpgContainer.GetEPGSearch(broadcasts, filter);
+
+  if (resultSize != 1)
+    return broadcast;
+
+  broadcast = broadcasts.Get(0);
+
+  return broadcast;
+}
+
+JSONRPC_STATUS CPVROperations::AddTimer(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  if (!g_PVRManager.IsStarted())
+    return FailedToExecute;
+
+  CFileItemPtr broadcast = CPVROperations::GetBroadcastFromBroadcastid(parameterObject["broadcastid"].asUnsignedInteger());
+  if (!broadcast)
+    return InvalidParams;
+
+  if (!broadcast->HasEPGInfoTag())
+    return InvalidParams;
+
+  CEpgInfoTagPtr epgTag = broadcast->GetEPGInfoTag();
+
+  if (!epgTag)
+    return InvalidParams;
+
+  bool repeating = parameterObject["repeating"].asBoolean(false);
+
+  if (epgTag->HasTimer())
+    return InvalidParams;
+
+  CPVRTimerInfoTagPtr newTimer = CPVRTimerInfoTag::CreateFromEpg(epgTag, repeating);
+  if (newTimer)
+  {
+    bool sentOkay = g_PVRTimers->AddTimer(newTimer);
+    if (sentOkay)
```

Comment:
Is this kodi style? (I personally don't like a main function hidden in an if statement, but will modify if it's more in line with the rest of style!